frontend/src/types/models.ts

```diff
@@ -1,11 +1,43 @@
 export interface User {
+  id: string;
   email: string;
   password: string;
+  userName?: string;
   phoneNumber: string;
   firstName: string;
   lastName: string;
+  nationality: string;
+  inNationalTeam: boolean;
   clubName: string;
   danRank: string;
-  underage?: boolean;
+  underage: boolean;
   guardiansEmail?: string;
 }
+
+export type PlayerColor = "red" | "white";
+
+export type PointType = "men" | "kote" | "do" | "tsuki" | "hansoku";
+
+export type MatchType = "group" | "playoff";
+
+export interface MatchPoint {
+  type: PointType;
+  timestamp: Date;
+}
+
+export interface MatchPlayer {
+  id: string;
+  points: MatchPoint[];
+  color: PlayerColor;
+}
+export interface Match {
+  id: string;
+  startTimestamp?: Date;
```

Comment:
The preferable id type would be ObjectIdString, but on the client side it doesn't matter.